rust/src/io/reader.rs

```diff
@@ -306,8 +325,24 @@ async fn read_batch(
         .then(|f| async move { read_array(reader, f, batch_id, params).await })
         .try_collect::<Vec<_>>()
         .await?;
+
+    let deletion_vector = match &reader.fragment {
+        Some(frag) => {
+            frag.deletion_vector(reader.object_reader.object_store())
+                .await
+        }
```

Comment:
I don't think this is the point where we want to read in the deletion vector. `read_batch()` is called for every 1024 rows in the file, so we are effectively reloading this data maybe thousands of time per file.

I think we would prefer to read the deletion vector during `FileReader::try_new_with_fragment` and store it next to the PageTable in `FileReader`. This also avoids the need to add an `object_store` field onto `ObjectReader`, which is a bit awkward since that struct is meant to hold a reference only to a single file.